### PR TITLE
ProxMox builder: Use ssh_host config as host for the communicator

### PR DESCRIPTION
**Pull request by zaenk**
When `ssh_host` config is present, then the builder gives this value to the communicator.

The original builder requires `qemu-guest-agent` package to be installed on the VM in order to determine it's IP address.

For environments with static addresses, the IP address of this VM is known in advance, and might already be configured in the template json, e.g. in boot command for Ubuntu.

### CODE REVIEW

**Comment by zaenk** [comment]
Added WIP status, because without `qemu-guest-agent` installed in guest the proxmox builder cannot stop the VM.

It is caused by the `Agent: "1"` config in [builder/proxmox/step_start_vm.go#27](https://github.com/hashicorp/packer/blob/master/builder/proxmox/step_start_vm.go#L27). If QEMU agent is enabled, then proxmox will use the agent for shutdown, but the command will timeout if it is not installed on the guest.

**Comment by zaenk** [comment]
This is my first PR, so I need some help. What's the next step to get this approved/merged?

**Comment by azr** [comment]
Howdy @carlpett ! I think you're the best choice to answer here ! 🙂

**Comment by carlpett** [comment]
Hey @azr! I'll take a look!

File: builder/proxmox/builder.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 		},
 		&communicator.StepConnect{
 			Config:    &b.config.Comm,
-			Host:      getVMIP,
+			Host:      commHost(b.config.Comm.SSHHost),
```
Comment (carlpett) [inline builder/proxmox/builder.go]: Don't we need to take the WinRM communicator into consideration here as well?

File: builder/proxmox/config.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ type Config struct {
 	NICs    []nicConfig  `mapstructure:"network_adapters"`
 	Disks   []diskConfig `mapstructure:"disks"`
 	ISOFile string       `mapstructure:"iso_file"`
+	Agent   string       `mapstructure:"qemu_agent"`
```
[the remark this answers is not in the document]
Comment (zaenk) [reply builder/proxmox/config.go]: I have thought about it but I haven't found any examples under builders for optional boolean with `true` default value. I've managed to solve it in https://github.com/hashicorp/packer/pull/7564/commits/02b6e7f3de97d12371dcbd776d8a345789fa95fa but not completely happy with this code... but it works.

File: builder/proxmox/builder.go (hunk 1 above)
Comment (zaenk) [reply builder/proxmox/builder.go]: Good catch! Fortunately `communicator.Config.Host()` takes care of it by default!

File: builder/proxmox/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlpett) [reply builder/proxmox/config.go]: :+1:

File: builder/proxmox/builder.go (hunk 1 above)
Comment (carlpett) [reply builder/proxmox/builder.go]: :+1:

**Comment by carlpett** [comment]
Nice fixes!

**Comment by zaenk** [comment]
Thank you!

**Comment by azr** [comment]
Nice work people, LGTM too 🙂

**Pull request merged**